force-app/main/default/classes/ApplicationSettingsController_TEST.cls

```diff
@@ -0,0 +1,108 @@
+private with sharing class ApplicationSettingsController_TEST {
+    /**************************************************************************************************************************************************
+     * @description Test method to verify that the locateApplicationSettingsVMapper method returns instance of locateApplicationSettingsVMapper.
+     *************************************************************************************************************************************************/
+    @isTest
+    private static void locateApplicationSettingsVMapper() {
+        Test.startTest();
+        ApplicationSettingsVMapper applicationSettingsVMapperInstance = ApplicationSettingsController.locateApplicationSettingsVMapper();
+        Test.stopTest();
+
+        System.assertEquals(
+            ApplicationSettingsVMapper.getInstance(),
+            applicationSettingsVMapperInstance,
+            'Should return the instance of locateApplicationSettingsVMapper'
+        );
+    }
+
+    /****************************************************************************************************************************
+     * @description Test method to verify that the getApplicationSettingsVModel method returns an ApplicationSettingsVModel
+     * representing Application Settings in Hierarchy Settings.
+     ****************************************************************************************************************************/
+    @isTest
+    private static void getRelationshipSettingsVModelValid() {
+        ApplicationSettingsVMapper.instance = new STUB_ApplicationSettingsVMapper();
+
+        Test.startTest();
+        ApplicationSettingsVModel applicationSettingsVModel = ApplicationSettingsController.getApplicationSettingsVModel();
+        Test.stopTest();
+
+        System.assertEquals(
+            true,
+            applicationSettingsVModel.enableApplicationWindowValidation,
+            'Selected value should be retrieved for application window validation'
+        );
+    }
+
+    /****************************************************************************************************************************
+     * @description Test method to verify that the getApplicationSettingsVModel method fails gracefully when an exception is
+     * encountered.
+     ****************************************************************************************************************************/
+    @isTest
+    private static void getRelationshipSettingsVModelException() {
+        ApplicationSettingsVMapper.instance = new STUB_ExceptionApplicationSettingsVMapper();
+
+        try {
+            Test.startTest();
+            ApplicationSettingsVModel applicationSettingsVModel = ApplicationSettingsController.getApplicationSettingsVModel();
+            Test.stopTest();
+        } catch (AuraHandledException e) {
+            System.assertEquals('Script-thrown exception', e.getMessage(), 'Should throw exception.');
+        }
```

Comment:
This method will pass w/o exception. Suggesting to store exception in the actual exception var and verify if outside of the try/catch block.
```suggestion

        Exception actualException;
        try {
            Test.startTest();
            ApplicationSettingsVModel applicationSettingsVModel = ApplicationSettingsController.getApplicationSettingsVModel();
            Test.stopTest();
        } catch (AuraHandledException e) {
             actualException = e;
        }
        
        System.assertNotEquals(null, actualException, 'Should throw an exception.');
        System.assertEquals('Script-thrown exception', actualException.getMessage(), 'Exception message should match.');
```

Reply:
The reason why a generic message 'Script-throw exception' is generated is that `AuraHandledException` message is not set. In the case you would like to return exception `e.getMessage()` to the calling component, then need to `setMessage` (see [AuraEnabledUtility](https://github.com/SalesforceFoundation/omega/blob/main/force-app/main/default/classes/support/AuraEnabledUtility.cls#L9) on details). @MatthewBlanski , @xinnanlisf

Reply:
I'm placing an alternative here for posterity. Having the assert cause a test failure will also meet the need.
```suggestion

        try {
            Test.startTest();
            ApplicationSettingsVModel applicationSettingsVModel = ApplicationSettingsController.getApplicationSettingsVModel();
            Test.stopTest();
            System.assert(false,'An AuraHandledException should be thrown.');
        } catch (AuraHandledException e) {
            System.assertEquals('Script-thrown exception', e.getMessage(), 'Should throw exception.');
        }
```